frontend/src/components/core/Sidebar/SidebarNav.test.tsx

```diff
@@ -215,27 +214,20 @@ describe("SidebarNav", () => {
     )
   })
 
-  it("does not render an icon when not expanded and not overflowing", () => {
-    const wrapper = shallow(
-      <SidebarNav {...getProps({ hasSidebarElements: true })} />
-    )
-    expect(
-      wrapper.find(StyledSidebarNavSeparatorContainer).find(Icon).exists()
-    ).toBe(false)
-  })
-
-  it("renders ExpandMore icon when not expanded and overflowing", () => {
-    mockUseIsOverflowing.mockReturnValueOnce(true)
+  it("renders the correct StyledSidebarNavButton when collapsed", () => {
     const wrapper = shallow(
       <SidebarNav {...getProps({ hasSidebarElements: true })} />
     )
 
     expect(
-      wrapper.find(StyledSidebarNavSeparatorContainer).find(Icon).props()
-    ).toHaveProperty("content", ExpandMore)
+      wrapper
+        .find(StyledSidebarNavSeparatorContainer)
+        .find(StyledSidebarNavButton)
+        .props()
+    ).toHaveProperty("isExpanded", false)
   })
```

Comment:
I think this test could be improved by also checking the text content of the button (`# More`) in the same way the expanded test below does.

Also, I'd like to see a separate test added to check that the number of additional pages show in the `# More` button is calculated correctly in a couple scenarios to avoid strange behavior like the `-1 More` button

Reply:
Fixed on [e225815](https://github.com/streamlit/streamlit/pull/6050/commits/e225815d0f1ae8fdcd5abab5a3e32bc99b56eb0b), and added the extra tests on [14a502d](https://github.com/streamlit/streamlit/pull/6050/commits/14a502de2931656e5c67bc48f74ef70b4130e182)